### PR TITLE
Code style and dependencies updates

### PR DESCRIPTION
**Pull request by mderazon**
hey @cguedes what's up ? Gonna use this library soon and thought it would be good to update it a little. no code changes for now, just updates and style and minor stuff

- style code with prettier (i'm a prettier addict) and added prettier test for npm test script.
- update all dependencies.
- remove beta mentions from readme - don't think it's necessary and it hurts adoption probably.
- added some fields on the package.json for better discoverability in npm.
- fixed some typos and formatting in readme

the tests fail because the auth json file doesn't authenticate anymore or something like that.

I don't have access to the google cloud platform any longer so I can't check why the authentication fails. can you please check that ?

I also think it would be good to give it a 1.0.0 version instead of the "unstable" sub 1.0.0 version.

let me know if you have any other comments.

### CODE REVIEW

**Comment by mderazon** [comment]
has conflicts for some reason 🤔 

I have no idea how that happened, will fix it soon

**Comment by cguedes** [comment]
Hi @mderazon, everything is great. Thanks.

Agree with you about beta, versioning. Let me look to prettier lib, but have heard good things about it.

Will also check what is happening with the auth account for tests.

File: package.json [hunk 1]
```diff
@@ -2,13 +2,19 @@
   "name": "node-sheets",
   "version": "0.3.1",
   "description": "Utility Wrapper for Google Spreadsheets",
-  "author": "Carlos Guedes <carlos@urbancups.com> (http://cupsapp.com)",
+  "author": "Carlos Guedes <carlos@urbancups.com> (https://cupsapp.com)",
```
[the remark this answers is not in the document]
Comment (cguedes) [reply package.json]: And update my website to https://cupsworks.com)

[the remark this answers is not in the document]
Comment (mderazon) [reply package.json]: Sure, but I don't know if you can have multiple authors so you can leave it on you

will update the website

File: README.md [hunk 2]
```diff
@@ -20,42 +16,43 @@ $ npm install node-sheets --save
 Example to retrieve data from [this google spreadsheet](https://docs.google.com/spreadsheets/d/1amfst1WVcQDntGe6walYt-4O5SCrHBD5WntbjhvfIm4) using ES7 async/await.
 
 ```javascript
-import Sheets from 'node-sheets'
+import Sheets from 'node-sheets';
```
[the remark this answers is not in the document]
Comment (mderazon) [reply README.md]: prettier does use semicolon as default 😱 

But it's configurable. Prefer I'd have it without semi ?

File: src/sheets.js [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
+import Q from 'q';
+import google from 'googleapis';
+import util from 'util';
+import zipObject from 'lodash.zipobject';
 
-const SCOPE_DRIVE_READ = 'https://www.googleapis.com/auth/drive.readonly'
-const SCOPE_SPREADSHEETS_READ = 'https://www.googleapis.com/auth/spreadsheets.readonly'
+const SCOPE_DRIVE_READ = 'https://www.googleapis.com/auth/drive.readonly';
+const SCOPE_SPREADSHEETS_READ =
+  'https://www.googleapis.com/auth/spreadsheets.readonly';
```
[the remark this answers is not in the document]
Comment (mderazon) [reply src/sheets.js]: Prettier auto thing

File: src/sheets.js [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
 
 export default class Sheets {
-
   /**
    * Create an instance of Sheets for a given google spreadsheet id
    */
-  constructor (spreadsheetId) {
+  constructor(spreadsheetId) {
```
[the remark this answers is not in the document]
Comment (mderazon) [reply src/sheets.js]: Yes :-)

**Comment by mderazon** [comment]
prettier might have some choices we're not used to but the autoformatting on save and not having to think of code style is very nice. I'm converted :-)

**Comment by cguedes** [comment]
I agree with you about automatically format the document on save. Will take a look at Prettier.

> The "author" is one person. "contributors" is an array of people. A "person" is an object with a "name" field and optionally "url" and "email", like this: - [source](https://docs.npmjs.com/files/package.json#people-fields-author-contributors)

You are right about the author. I was thinking of contributors instead.

**Comment by mderazon** [comment]
JWT authentication fails for some reason and that's why it the test fail.

```
Error: invalid_grant: Invalid JWT Signature.
    at /Users/me/home/node-sheets/node_modules/gtoken/src/index.ts:249:19
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cguedes) [reply README.md]: Yes. But here (prettier) we can live with them.
With the VS Code auto format active (will add the setting to .vscode/settings.json) this shouldn't be a problem.

File: package.json [hunk 5]
```diff
@@ -21,23 +36,24 @@
     "lib"
   ],
   "dependencies": {
-    "babel-runtime": "^6.11.6",
-    "google-auth-library": "^0.9.8",
-    "googleapis": "^12.4.0",
+    "babel-runtime": "^6.26.0",
+    "google-auth-library": "^1.1.0",
+    "googleapis": "^25.0.0",
```
Comment (cguedes) [inline package.json]: This was breaking (most of) the tests. In version 25.x the google API is not returning an array, but instead an object with .data with the payload.

Comment (cguedes) [reply package.json]: Will fix in the next commit.

Comment (cguedes) [reply package.json]: Here is the page describing the change: https://github.com/google/google-api-nodejs-client/blob/474bed78474516ecbb3774f42712063c847f86e8/MIGRATING.md

**Pull request merged**